instrumentation/trace-annotation/src/main/java/io/opentelemetry/auto/instrumentation/traceannotation/TraceDecorator.java

```diff
@@ -15,13 +15,66 @@
  */
 package io.opentelemetry.auto.instrumentation.traceannotation;
 
+import static net.bytebuddy.matcher.ElementMatchers.isDeclaredBy;
+import static net.bytebuddy.matcher.ElementMatchers.none;
+
 import io.opentelemetry.OpenTelemetry;
 import io.opentelemetry.auto.bootstrap.instrumentation.decorator.BaseDecorator;
+import io.opentelemetry.auto.config.Config;
+import io.opentelemetry.contrib.auto.annotations.WithSpan;
 import io.opentelemetry.trace.Tracer;
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.Set;
+import net.bytebuddy.description.ByteCodeElement;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
 
 public class TraceDecorator extends BaseDecorator {
   public static final TraceDecorator DECORATE = new TraceDecorator();
 
   public static final Tracer TRACER =
       OpenTelemetry.getTracerProvider().get("io.opentelemetry.auto.trace-annotation");
+
+  /**
+   * This method is used to generate an acceptable span (operation) name based on a given method
+   * reference. It first checks for existence of {@link WithSpan} annotation. If it is present, then
+   * tries to derive name from its {@code value} attribute. Otherwise delegates to {@link
+   * #spanNameForMethod(Method)}.
+   */
+  public String spanNameForMethodWithAnnotation(final Method method) {
+    WithSpan annotation = method.getAnnotation(WithSpan.class);
+    if (annotation != null && !annotation.value().isEmpty()) {
+      return annotation.value();
+    }
+
+    return spanNameForMethod(method);
+  }
+
+  /*
+  Returns a matcher for all methods that should be excluded from auto-instrumentation by
+  annotation-based advices.
+   */
+  static ElementMatcher.Junction<MethodDescription> configureExcludedMethods() {
```

Comment:
I think the cause of the muzzle failures is this method. I don't think the classloader that the advice classes are applied to has bytebuddy in it's classpath

Reply:
Looks like you moved it here from one of the Instrumenter classes, might want to make an abstract parent class if this is needed multiple places

Reply:
Thanks!